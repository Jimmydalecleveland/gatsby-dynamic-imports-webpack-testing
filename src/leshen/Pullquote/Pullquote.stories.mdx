import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Pullquote, { ticketID } from '.'
import Section from '../Section'
import VariableContent from '../VariableContent'
import Carousel from '../Carousel'
import Typography from '../Typography'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|Pullquote' component={Pullquote} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Pullquote

<Preview>
  <Story name="default">
    <VariableContent centeredContent>
      <Typography>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo dolores
        laboriosam, consequatur molestiae ea hic beatae, numquam quis maiores
        aliquam incidunt culpa repudiandae nostrum enim rem itaque! Dolore,
        impedit possimus.
      </Typography>
      <Pullquote cite="John Doe">
        &quot;Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
        quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
        molestias recusandae tenetur fuga ipsam enim.&quot;
      </Pullquote>
      <Typography>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo dolores
        laboriosam, consequatur molestiae ea hic beatae, numquam quis maiores
        aliquam incidunt culpa repudiandae nostrum enim rem itaque! Dolore,
        impedit possimus.
      </Typography>
    </VariableContent>
  </Story>
</Preview>

## Props
<Props of={Pullquote} />

## In a Carousel

<Preview>
  <Story name="in a carousel">
    <>
      <Section>
        <Carousel
          dots
          infinite={false}
          speed={500}
          slidesToShow={1}
          slidesToScroll={1}
          initialSlide={0}
        >
          <StyledPullquoteContainer key={1}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={2}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid
              fuga eligendi debitis deserunt commodi maxime culpa. Reiciendis
              doloremque unde molestias fuga et id? Iusto, ex?
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={3}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={4}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={5}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
        </Carousel>
      </Section>
      <Section prominent>
        <Carousel
          dots
          infinite={false}
          speed={500}
          slidesToShow={1}
          slidesToScroll={1}
          initialSlide={0}
        >
          <StyledPullquoteContainer key={1}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={2}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid
              fuga eligendi debitis deserunt commodi maxime culpa. Reiciendis
              doloremque unde molestias fuga et id? Iusto, ex?
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={3}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={4}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
          <StyledPullquoteContainer key={5}>
            <Pullquote cite="John Doe">
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus
              quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus
              molestias recusandae tenetur fuga ipsam enim.
            </Pullquote>
          </StyledPullquoteContainer>
        </Carousel>
      </Section>
    </>
  </Story>
</Preview>
