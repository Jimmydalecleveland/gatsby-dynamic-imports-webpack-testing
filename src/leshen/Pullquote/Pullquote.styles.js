import styled from '@emotion/styled'
import Typography from '../Typography'

export const Wrapper = styled.div`
  margin: ${({ theme }) =>
    `${theme.spacing.md} ${theme.spacing.lg} ${theme.spacing.md}`};

  @media (min-width: ${({ theme }) => theme.breakpoints.md}) {
    margin: ${({ theme }) =>
      `${theme.spacing.lg} ${theme.spacing.lg} ${theme.spacing.md}`};
  }
`

export const Cite = styled(Typography)`
  margin-top: ${({ theme }) => theme.spacing.sm};
  display: flex;
  justify-content: center;
  align-items: center;

  &::before,
  &::after {
    margin-right: ${({ theme }) => theme.spacing.xs};
    content: '';
    border-bottom: solid 1px;
    width: 20px;
  }

  &::after {
    margin-left: ${({ theme }) => theme.spacing.xs};
    margin-right: 0;
  }
`
