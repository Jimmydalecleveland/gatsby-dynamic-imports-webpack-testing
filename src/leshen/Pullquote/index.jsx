import React from 'react'
import PropTypes from 'prop-types'
import * as Styled from './Pullquote.styles'

import Typography from '../Typography'

const Pullquote = ({ children, cite }) => (
  <Styled.Wrapper>
    <Typography variant="h4" rendersAs="blockquote">
      {children}
    </Typography>
    {cite && (
      <Styled.Cite variant="feature" rendersAs="span">
        {cite}
      </Styled.Cite>
    )}
  </Styled.Wrapper>
)

Pullquote.defaultProps = {
  cite: '',
}

Pullquote.propTypes = {
  children: PropTypes.string.isRequired,
  cite: PropTypes.string,
}

export default Pullquote

export const ticketID = 'CMS-173'
