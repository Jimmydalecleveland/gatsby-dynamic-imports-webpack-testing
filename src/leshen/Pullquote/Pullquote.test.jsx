import React from "react"
// import fluidRange from '../utils/fluidRange'
import { render } from "setup-tests"
import Pullquote from "."
// import theme from '../utils/theme'

describe("Pullquote", () => {
  test("should render", () => {
    const { container } = render(
      <Pullquote>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua.
      </Pullquote>
    )
    expect(container).toMatchSnapshot()

    // TODO: fluidRange currently doesn't work because it returns scss which causes the parser to fail
    // Make sure that it renders as h4's styles
    // expect(container.firstChild.firstChild).toHaveStyle(`
    //   ${fluidRange('font-size', theme.fonts.subheadings.lg.size)}
    //   font-weight: ${theme.fonts.subheadings.lg.weight};
    //   ${fluidRange('line-height', theme.fonts.subheadings.lg.lineHeight)}
    // `)
  })

  test("should render with optional citation", () => {
    const { container } = render(
      <Pullquote cite="John Doe">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus quis
        vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus molestias
        recusandae tenetur fuga ipsam enim.
      </Pullquote>
    )
    expect(container).toMatchSnapshot()
  })
})
