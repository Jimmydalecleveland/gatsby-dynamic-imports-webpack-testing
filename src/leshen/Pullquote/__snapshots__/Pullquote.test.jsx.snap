// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pullquote should render 1`] = `
.emotion-2 {
  margin: 24px 32px 24px;
}

@media (min-width:768px) {
  .emotion-2 {
    margin: 32px 32px 24px;
  }
}

.emotion-0 {
  font-weight: 500;
  font-size: calc(0.8333333333333334vw + 18px);
  line-height: calc(1.6666666666666667vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 22px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 36px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <blockquote
      class="emotion-0 emotion-1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </blockquote>
    
  </div>
</div>
`;

exports[`Pullquote should render with optional citation 1`] = `
.emotion-5 {
  margin: 24px 32px 24px;
}

@media (min-width:768px) {
  .emotion-5 {
    margin: 32px 32px 24px;
  }
}

.emotion-0 {
  font-weight: 500;
  font-size: calc(0.8333333333333334vw + 18px);
  line-height: calc(1.6666666666666667vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 22px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 36px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-3 {
  font-size: 1.125rem;
  line-height: 1.5rem;
  margin-top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-3 {
    width: 100%;
  }
}

.emotion-3::before,
.emotion-3::after {
  margin-right: 12px;
  content: '';
  border-bottom: solid 1px;
  width: 20px;
}

.emotion-3::after {
  margin-left: 12px;
  margin-right: 0;
}

<div>
  <div
    class="emotion-5 emotion-6"
  >
    <blockquote
      class="emotion-0 emotion-1"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Delectus quis vero ad deleniti iusto quo sequi omnis? Cupiditate doloribus molestias recusandae tenetur fuga ipsam enim.
    </blockquote>
    <span
      class="emotion-2 emotion-3 emotion-1"
    >
      John Doe
    </span>
  </div>
</div>
`;
